package org.example.azoi.model;

import jakarta.persistence.*;

@Entity
@Table(name = "roles", schema = "azoi")
public class Role {

    public static final Role ROLE_USER = new Role("ROLE_USER", "普通用户");
    public static final Role ROLE_ADMIN = new Role("ROLE_ADMIN", "管理员");

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @Column(name = "id", nullable = false)
    private Long id;

    @Column(name = "code", nullable = false, length = 32)
    private String code;

    @Column(name = "name", nullable = false, length = 64)
    private String name;

    public Role() {
    }

    public Role(String code, String name) {
        this.code = code;
        this.name = name;
    }

    public Long getId() {
        return id;
    }

    public void setId(Long id) {
        this.id = id;
    }

    public String getCode() {
        return code;
    }

    public void setCode(String code) {
        this.code = code;
    }

    public String getName() {
        return name;
    }

    public void setName(String name) {
        this.name = name;
    }

}